docs/collecting-data/collecting-from-own-applications/javascript-trackers/browser-tracker/browser-tracker-v3-reference/advanced-usage/using-an-id-service/index.md

```diff
@@ -0,0 +1,154 @@
+
+As of Safari 16.4 released in April 2023, Safari sets the lifetime of server-set first-party cookies to a maximum of 7 days in the following cases:
+
+1. The server setting the cookie is behind a CNAME that resolves (at any point) to a host that is third-party to the website the user is currently browsing.
+2. The server setting the cookie is set with A/AAAA records that resolve to an IP address (IP4 or IP6) where the first half of the address does not match the first half of the IP address for the server on the website the user is currently browsing. (e.g. 203.52.1.2 and 203.52.56.22 are okay, 201.55.1.2 is not).
+
+This greatly limits the effectiveness of tracking a customer journey where users are not regularly logging into your website.  Without a strong understanding of your customers, downstream use cases including marketing attribution, product analytics and personalized recommendations are difficult to achieve.
+
+**The ID service can help enhance the confidence in persistent browser identifiers that Snowplow tracking provides and specifically for this guide, the `network_userid`.**
+
+## Developing and deploying an ID service
+
+The ID service is code that needs to be deployed on and executed from the same IP space that serves the main web document of your application. This is probably the web application system or the CDN in front of the application.
+
+This code has minimal functionality and based on our experience can either be:
+
+- An **API endpoint** that is part of the main web application.
+- **A function running in an edge worker.** Almost all modern CDNs allow the addition of ‘middleware’ code which runs on requests at edge nodes in front of the client’s application domain.
+- **A custom middleware based on the customer’s framework**. E.g. ExpressJS middleware, Next.js middleware, Play custom action etc. which can run on every document request.
+- **A low-footprint application with a single endpoint**. A Go web server or something along these lines.
+
+### Developing the ID service code
+
+The responsibilities of this service are:
+1. Create a unique identifier for this browser, set it in a cookie and return it in a `Set-Cookie` header on a domain **accessible** by the Snowplow collector deployed domain. Other cookie header attributes, such as `Domain`, `Secure`, `SameSite`, `httpOnly` etc., should have exactly the same values as the ones set on the collector configuration.
+2. Set the cookie used for the `network_userid` identifier, by default a cookie named `sp` or the value used in `collector.config.name` for the Snowplow collector, to the same value.
+
+_The new unique identifier cookie for sake of simplicity in this document will have the name `spIdService`._
+
+:::tip
+
+Examples showcasing an **accessible** domain in the `Set-Cookie` header for the new identifier:
+
+```js
+/* Collector c.customer.com - Domain customer.com */
+Set-Cookie: spIdService=12345; Domain=customer.com; Secure; HttpOnly
```

Comment:
We should set `Path=/` here or nothing will work.

Reply:
Thanks for catching that, it was written by hand!